AppCenterDistribute/AppCenterDistributeTests/MSDistributeCheckForUpdateTests.m

```diff
@@ -3,112 +3,283 @@
+  OCMStub([distributeMock openUrlInAuthenticationSessionOrSafari:OCMOCK_ANY]).andDo(nil);
+  XCTestExpectation *expectation = [self expectationWithDescription:@"Start update processed"];
 
   // When
-  [distributeMock checkForUpdateWithUpdateToken:updateToken distributionGroupId:distributionGroupId releaseHash:releaseHash];
+  [MSDistribute disableAutomaticCheckForUpdate];
+  MSDistribute.updateTrack = MSUpdateTrackPrivate;
+  [MSDistribute checkForUpdate];
+  dispatch_async(dispatch_get_main_queue(), ^{
+    [expectation fulfill];
+  });
 
   // Then
-  OCMVerify([distributeMock checkLatestRelease:updateToken distributionGroupId:distributionGroupId releaseHash:releaseHash]);
+  [self waitForExpectationsWithTimeout:1
+                               handler:^(NSError *error) {
+                                 // Then
+                                 OCMVerify([distributeMock startUpdateOnStart:NO]);
+                                 OCMVerify([distributeMock openUrlInAuthenticationSessionOrSafari:OCMOCK_ANY]);
+                                 if (error) {
+                                   XCTFail(@"Expectation Failed with error: %@", error);
+                                 }
+                               }];
 
   // Cleanup
   [distributeMock stopMocking];
 }
 
-- (void)testCheckForUpdateDoesNotCheckWhenDisabled {
+- (void)testCheckForUpdateGetsLatestReleaseEvenThoughAutomaticUpdateIsDisabled {
 
   // If
-  NSString *updateToken = @"UpdateToken";
-  NSString *distributionGroupId = @"DistributionGroupId";
-  NSString *releaseHash = @"ReleaseHash";
-  MSDistribute *distribute = [MSDistribute sharedInstance];
-  id distributeMock = OCMPartialMock(distribute);
-  [distributeMock setValue:@(YES) forKey:@"updateFlowInProgress"];
-  [distributeMock setValue:@(YES) forKey:@"automaticCheckForUpdateDisabled"];
-  OCMReject([distributeMock checkLatestRelease:updateToken distributionGroupId:distributionGroupId releaseHash:releaseHash]);
+  NSDictionary<NSString *, id> *plist = @{@"CFBundleShortVersionString" : @"1.0", @"CFBundleVersion" : @"1"};
+  OCMStub([self.bundleMock infoDictionary]).andReturn(plist);
+
+  // Distribute Mock
+  __block id distributeMock = OCMPartialMock(self.sut);
+  OCMStub([distributeMock canBeUsed]).andReturn(YES);
+  OCMStub([distributeMock isEnabled]).andReturn(YES);
+  [self.settingsMock removeObjectForKey:kMSUpdateTokenRequestIdKey];
+  OCMStub([distributeMock checkForUpdatesAllowed]).andReturn(YES);
+  XCTestExpectation *expectation = [self expectationWithDescription:@"Start update processed"];
+
+  // Ingestion Mock
+  __block id ingestionMock = OCMClassMock([MSDistributeIngestion class]);
+  OCMStub([ingestionMock checkForPublicUpdateWithQueryStrings:OCMOCK_ANY completionHandler:OCMOCK_ANY]).andDo(^(NSInvocation *invocation) {
+    void (^handler)(NSString *callId, NSHTTPURLResponse *_Nullable response, NSData *_Nullable data, NSError *_Nullable error);
```

Comment:
Please add typedef to make it more readable

Reply:
We don't do this in unit tests and just one time use only.